src/app/common/hooks/use-convert-to-fiat-amount.ts

```diff
@@ -2,29 +2,64 @@ import { useCallback, useMemo } from 'react';
 
 import { CryptoCurrencies } from '@shared/models/currencies.model';
 import { createMarketData, createMarketPair } from '@shared/models/market.model';
-import type { Money } from '@shared/models/money.model';
+import { type Money, createMoney } from '@shared/models/money.model';
+import { isUndefined } from '@shared/utils';
 
-import { useCryptoCurrencyMarketData } from '@app/query/common/market-data/market-data.hooks';
+import { i18nFormatCurrency } from '@app/common/money/format-money';
+import { unitToFractionalUnit } from '@app/common/money/unit-conversion';
+import { useCryptoCurrencyMarketDataMean } from '@app/query/common/market-data/market-data.hooks';
 
 import { baseCurrencyAmountInQuote } from '../money/calculate-money';
 
-export function useConvertCryptoCurrencyToFiatAmount(currency: CryptoCurrencies) {
-  const cryptoCurrencyMarketData = useCryptoCurrencyMarketData(currency);
+export function useConvertCryptoCurrencyToFiatAmountMean(currency: CryptoCurrencies) {
```

Comment:
Maybe when working on this we could refactor some hooks to be standard functions? 

I could be mistaken here but looking at the code I think in some cases they don't really need to be hooks and could instead receive their data.

That may make it easier to add better tests